src/thunderbots/software/network_input/main.cpp

```diff
@@ -49,17 +31,80 @@ int main(int argc, char** argv)
     // Initialize the logger
     Util::Logger::LoggerSingleton::initializeLogger(node_handle);
 
+    // Init our backend class
+    Backend backend = Backend();
+
+    // Store the state of the most up to date World message
+    thunderbots_msgs::World world_msg;
+
+    // Create the io_service that will be used to service all network requests
+    boost::asio::io_service io_service;
+
+    // The function that will be run on every received SSL_WrapperPacket
+    auto filter_and_publish_vision_data = [&backend, world_publisher,
+                                           &world_msg](SSL_WrapperPacket packet) {
```

Comment:
This should not be a lambda function. It's _way_ too big.

Reply:
Refactored to wrap the clients and lambda functions in a general `NetworkClient`